sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -1491,12 +1494,36 @@ private static BigQueryWriteClient newBigQueryWriteClient(BigQueryOptions option
       return BigQueryWriteClient.create(
           BigQueryWriteSettings.newBuilder()
               .setCredentialsProvider(() -> options.as(GcpOptions.class).getGcpCredential())
+              .setBackgroundExecutorProvider(new OptionsExecutionProvider(options))
               .build());
     } catch (Exception e) {
       throw new RuntimeException(e);
     }
   }
 
+  /**
+   * OptionsExecutionProvider is a utility class used to wrap the Pipeline-wide {@link
+   * ScheduledExecutorService} into a supplier for the {@link BigQueryWriteClient}.
+   */
+  private static class OptionsExecutionProvider implements ExecutorProvider {
```

Comment:
Consider using https://github.com/googleapis/gax-java/blob/47b9247a50347e9fd61ad5a9f5adaa4bdb757a26/gax/src/main/java/com/google/api/gax/core/FixedExecutorProvider.java#L54 instead of creating a class that is effectively a FixedExecutorProvider